swap/src/network/transport.rs

```diff
@@ -21,9 +23,12 @@ pub fn build(id_keys: &identity::Keypair) -> Result<SwapTransport> {
     let noise = NoiseConfig::xx(dh_keys).into_authenticated();
 
     let tcp = TokioTcpConfig::new().nodelay(true);
-    let dns = TokioDnsConfig::system(tcp)?;
+    let dns = TokioDnsConfig::system(tcp.clone())?;
 
-    let transport = dns
+    let websocket = WsConfig::new(dns);
+
+    let transport = websocket
+        .or_transport(tcp)
```

Comment:
Sorry, I didn't see this in my initial review but I think we might actually want to clone the `dns` transport instead of the `tcp` one. Otherwise we lose DNS support outside of the Websocket transport.

Reply:
Sorry to only mention this now, didn't occur to me in my initial review!